src/openrct2/ride/Ride.cpp

```diff
@@ -4097,6 +4097,9 @@ static Vehicle* vehicle_create_car(
     if (vehicle == nullptr)
         return nullptr;
 
+    if (trackElement == nullptr)
+        return nullptr;
+
```

Comment:
Small possible optimization, but we should probably do this check first in the function, then we avoid calling a handful of gets?